builtin/providers/librato/resource_librato_metric.go

```diff
@@ -0,0 +1,432 @@
+
+	_, err = client.Metrics.Edit(metric)
+	if err != nil {
+		return fmt.Errorf("Error updating Librato metric: %s", err)
+	}
+
+	log.Printf("[INFO] Updated Librato metric %s", metricID)
+
+	// Wait for propagation since Librato updates are eventually consistent
+	wait := resource.StateChangeConf{
+		Pending:                   []string{fmt.Sprintf("%t", false)},
+		Target:                    []string{fmt.Sprintf("%t", true)},
+		Timeout:                   5 * time.Minute,
+		MinTimeout:                2 * time.Second,
+		ContinuousTargetOccurence: 5,
+		Refresh: func() (interface{}, string, error) {
+			log.Printf("[INFO] Checking if Librato Metric %s was updated yet", metricID)
+			changedMetric, _, getErr := client.Metrics.Get(metricID)
+			if getErr != nil {
+				return changedMetric, "", getErr
+			}
+			isEqual := reflect.DeepEqual(*fullMetric, *changedMetric)
+			log.Printf("[INFO] Updated Librato Metric %s match: %t", metricID, isEqual)
+			return changedMetric, fmt.Sprintf("%t", isEqual), nil
+		},
+	}
+
+	_, err = wait.WaitForState()
+	if err != nil {
+		log.Printf("[INFO] ERROR - Failed updating Librato Metric %s: %s", metricID, err)
+		return fmt.Errorf("Failed updating Librato Metric %s: %s", metricID, err)
+	}
+
+	return resourceLibratoMetricRead(d, meta)
+}
+
+func resourceLibratoMetricDelete(d *schema.ResourceData, meta interface{}) error {
+	client := meta.(*librato.Client)
+
+	id := d.Id()
+
+	log.Printf("[INFO] Deleting Metric: %s", id)
+	_, err := client.Metrics.Delete(id)
+	if err != nil {
+		return fmt.Errorf("Error deleting Metric: %s", err)
+	}
+
+	log.Printf("[INFO] Verifying Metric %s deleted", id)
+	resource.Retry(1*time.Minute, func() *resource.RetryError {
+
+		log.Printf("[INFO] GETing Metric %s", id)
+		_, _, err := client.Metrics.Get(id)
+		if err != nil {
+			if errResp, ok := err.(*librato.ErrorResponse); ok && errResp.Response.StatusCode == 404 {
+				log.Printf("[INFO] GET returned a 404 for Metric %s\n", id)
+				return nil
+			}
+			log.Printf("[INFO] non-retryable error attempting to Get metric: %s", err)
+			return resource.NonRetryableError(err)
+		}
+
+		log.Printf("[INFO] retryable error attempting to Get metric: %s", id)
+		return resource.RetryableError(fmt.Errorf("metric still exists"))
+	})
+
+	log.Println("[INFO] I think Metric is deleted")
+
+	d.SetId("")
+	return nil
+}
+
+func metricReadResult(d *schema.ResourceData, metric *librato.Metric) error {
+	d.SetId(*metric.Name)
+	d.Set("id", *metric.Name)
+	d.Set("name", *metric.Name)
+	d.Set("type", *metric.Type)
+
+	if metric.Description != nil {
+		d.Set("description", *metric.Description)
+	}
+
+	if metric.DisplayName != nil {
+		d.Set("display_name", *metric.DisplayName)
+	}
+
+	if metric.Period != nil {
+		d.Set("period", *metric.Period)
+	}
+
+	if metric.Composite != nil {
+		d.Set("composite", *metric.Composite)
+	}
+
+	attributes := metricAttributesGather(d, metric.Attributes)
+	d.Set("attributes", attributes)
```

Comment:
Since attributes isn't a simple terraform type (`TypeList`), it's best to catch the error returned from the `d.Set()` function, and handle accordingly. 

```
if err := d.Set("attributes", attributes); err != nil {
  return err
}
```